xdsl/transforms/canonicalization_patterns/riscv.py

```diff
@@ -129,6 +129,24 @@ def match_and_rewrite(self, op: riscv.AddiOp, rewriter: PatternRewriter) -> None
             rewriter.replace_matched_op(riscv.MVOp(op.rs1, rd=rd))
 
 
+class AddImmediateConstant(RewritePattern):
+    @op_type_rewrite_pattern
+    def match_and_rewrite(self, op: riscv.AddiOp, rewriter: PatternRewriter) -> None:
+        if (
+            isinstance(li := op.rs1.owner, riscv.LiOp)
```

Comment:
Please remind me why do we do `.owner` here instead of checking if it's an instance of `OpResult` (like in other canonicalizations here)?

Reply:
It avoids an extra checks, I think the other rewrites should actually be changed, but it's not a high priority.

Reply:
Yeah, OK, apart from saving a check, what is the difference?